### ISSUE TITLE
Reolink Argus Eco - PIR Sensor/SDCard download

### ISSUE THREAD

**Issue by ment0s84** [thirtythreeforty/neolink#248]
Hello,
I just bought a "Reolink Argus Eco" camera (with battery) from the manufacturer's website. I hadn't really read the documentation stating that RTSP / HTTPS was not available with this hardware.

Thanks to the "Neolink" application, I manage to connect to the camera in "RTSP" and I can "easily" record and view the camera from my virtual server (Debian) with the FFMPEG and FFPLAY softwares.

However, I will wish:

- trigger my scripts (RTSP / Neolink, FFMPEG and FFPLAY) only when the camera detects a presence with the "PIR" sensor.
- retrieve the contents stored from the SD card from a shell script.

How to proceed? Are any of my wishes included in the Neolink Roadmap, and if so, do you have any idea about the duration of "delivery"?

I would also like to thank you for this very clean and efficient plugin.

Here is the configuration I use so far:

- Camera: Reolink Argus Eco + solar panel
- Server :
-- Debian on VirtualBox
-- Neolink for the RTSP protocol
-- FFMPEG for recording video streams
-- FFPLAY to display the video stream 

PS: Sorry, but Google Translate is working for me :).

**Comment by QuantumEntangledAndy** [comment]
We have motion detection setup on the mqtt branch already. You can use that to listen to motion detection messages. I'm not sure if motion detection is trigged in same way as pir because I don't one a camera that supports it so I've never had a chance to reverse engineer their messages.

Nothing so far for downing off of the camera. Even if you did it might not be as useful as you'd expect because the files are written in their own custom format. You'd have to decode this format. I think @twistedddx has a program for converting them. This sort of thing is something I'd like to add one day but there are many other things that need doing too PRs are welcome.

**Comment by QuantumEntangledAndy** [comment]
You can find our builds for other branches under actions. But it can be a bit unclear how to navigate them. The last mqtt build was here https://github.com/QuantumEntangledAndy/neolink/actions/runs/1132360541. But looking at it the builds have expired (GitHub only stores the binaries for so long). I will try to make a more recent build for you when I have time and merge in any newer features into it like udp.

**Comment by MasterPlexus** [comment]
> Even if you did it might not be as useful as you'd expect because the files are written in their own custom format.

I'm not sure on this. At least on Argus 2 I could say the Files are written in standard fat32 file system as MP4 files (one for substream, and one for Main). But I fear a kind of File-Access is not established (FTP or whatever) to made them directly available.
Sadly what they did with this great devices

**Comment by QuantumEntangledAndy** [comment]
I see perhaps they are more accessible then they were in previous models.

The bc api (their own custom one that we are reverse engineering) has a means to access the files. I've thought about maybe making that accessible over fuse a few times. It's on the todo list but like many things with the project it's all done only when we've got time.

**Comment by ment0s84** [comment]
Thank you for these answers.

I confirm today that it is impossible to access the data in FTP, which is a shame because the camera is of good quality (battery, sensor, resolution, etc.).

The last time I developed was 15 years ago, I will not be able to help you with the "reverse engineering", but I will follow your developments carefully :).

**Comment by QuantumEntangledAndy** [comment]
I've just merged master into the mqtt_cmd branch. That should hopefully produce binaries for you [here](https://github.com/QuantumEntangledAndy/neolink/actions/runs/1707080728) (It will take a few hours to finish building and they will be at the bottom of the page)

**Comment by juanaviladev** [comment]
> I've just merged master into the mqtt_cmd branch. That should hopefully produce binaries for you [here](https://github.com/QuantumEntangledAndy/neolink/actions/runs/1707080728) (It will take a few hours to finish building and they will be at the bottom of the page)

It seems there are some conflicts and the build has failed. Thank you for your amazing work!!

**Comment by QuantumEntangledAndy** [comment]
Ok next build should be [here](https://github.com/QuantumEntangledAndy/neolink/actions/runs/1708326255) I am going to get some sleep now so if it's still not working post it here and I'll poke it again :)

**Comment by daza67** [comment]
Did anyone get the MQTT side working with the mqtt_cmd branch ?  I compiled the latest commit successfully but is throwing an the error below. Also using an Argus Eco, would be great to get PIR updates via MQTT, battery status would also be nice.
thread 'main' panicked at 'Non-unique argument name: config is already in use',  /home/xxxxxx/.cargo/registry/src/github.com-1ecc6299db9ec823/clap-2.33.3/src/app/parser.rs:181:9"

**Comment by QuantumEntangledAndy** [comment]
Yes it compiled fine and their is a build with the binaries in the link I provided in my last post. [here](https://github.com/QuantumEntangledAndy/neolink/actions/runs/1708326255)

Also `Non-unique argument name: config is already in use` just means that your command line is not right. What did you call it with.

**Comment by QuantumEntangledAndy** [comment]
I don't own a battery camera so I cannot reverse engineer anything to do with battery camera's without someone doing most of the wireshark legwork. I also do not have PIR, but it might be the same message as motion detected which the current MQTT can handle.

**Comment by daza67** [comment]
Thanks, I did get it working with your build binaries, they weren't immediately obvious to download for me (I'm new to github). I do see connected and offline messages sent when I start and stop neolink however I don't seem to see any messages in response to the PIR triggering. So as you mention the messaging could be different, I'll work on getting some wireshark captures next.

**Comment by daza67** [comment]
Here is a wireshark capture of a motion event.
[home - switch_MS-00_18_0a_b3_eb_d2_IF-8.zip](https://github.com/thirtythreeforty/neolink/files/8240761/home.-.switch_MS-00_18_0a_b3_eb_d2_IF-8.zip)

**Comment by BeanBagKing** [comment]
@QuantumEntangledAndy / @thirtythreeforty  - I'm trying to get this working with my battery powered cameras as well. Constant RTSP seems to work fine, but this obviously isn't desirable as it's going to wear down the battery quickly. What I wanted, and what seems to be the request here, is to either a) only start streaming when the PIR sensor turns on (and stop streaming at some point), or b) don't stream, but rather let the camera write the video to the SD card, and then pick that up and "replay" it as an RTSP stream for BI. I'm mostly stating this to make sure I understand the current state of the project.

If this is right, and you think you could get it working if you had a battery powered camera to reverse engineer, I might be able to send you one, assuming shipping isn't crazy. It would be cheaper than having to purchase new/compatible equipment and would help out everyone :P It would likely be a refurb Argus Pro since they're fairly cheap and that's what I'm working with. If you're interested, let me know how I can get in touch for shipping.
